database/postgres/event_delivery.go

```diff
@@ -370,7 +370,7 @@ func (e *eventDeliveryRepo) FindEventDeliveryByID(ctx context.Context, projectID
 
 func (e *eventDeliveryRepo) FindEventDeliveryByIDSlim(ctx context.Context, projectID string, id string) (*datastore.EventDelivery, error) {
 	eventDelivery := &datastore.EventDelivery{}
-	err := e.db.QueryRowxContext(ctx, fetchEventDeliverySlim, projectID, id).StructScan(eventDelivery)
+	err := e.db.GetDB().QueryRowxContext(ctx, fetchEventDeliverySlim, projectID, id).StructScan(eventDelivery)
```

Comment:
```suggestion
	err := e.db.GetReadDB().QueryRowxContext(ctx, fetchEventDeliverySlim, projectID, id).StructScan(eventDelivery)
```

Reply:
This real-time method is used by `ProcessEventDelivery` which returns `DeliveryError` if not found. Will this be retried if not found on the replica?